### ISSUE TITLE
[Experimental] Dry-run Authorization Policy

### ISSUE THREAD

**Issue by yangminzhu** [istio/enhancements#25]
[//]: # (The syntax preceeding this line is a comment marker used to help guide the issue author with submission of an issue)
[//]: # (to github. Unlike HTML comments commonly used throughout istio.io documentation, this comment will not be rendered)
[//]: # (by github. Comments must be separated by carriage return preceding and concluding the text and be a single line.)

[//]: # (This page lists the requirements for an experimental feature. Completion of this template enables Istio work groups)
[//]: # (to collect information on potential new functionality. This template should be completed before users are exposed to)
[//]: # (any new experimental feature. Please complete this template during development.)

[//]: # (The feature implementation section must be completed before submission of the issue.)

# Feature:

[//]: # (All information in this section is mandatory for new issue submission.)

**Feature name:** Dry-run authorization policy

[//]: # (The name of the feature, e.g. Multiple control planes)

**Primary lead(s):** @yangminzhu 

[//]: # (The primary lead or leads responsible for the feature. These individuals serve as a point of contact for the feature.)

**Short description:** Allow customers to dry-run an authorization policy to test the effect using real traffic without enforcing the policy, reducing the risk of creating or changing the authorization policy.

[//]: # (A short description of the feature. One or two sentences maximum.)

# Requirements:

[//]: # (All information in this section is mandatory for promotion. Please edit the issue to modify the links in this)
[//]: # (section.)

- [x] [User stories](insert_your_link_here) reviewed in a work group meeting. (presented in the Security WG on Mar 17, 2021)

[//]: # (User stories are a way to communicate user value. User stories follow the style)
[//]: # (as a [type of user], I want [an action] so that [a benefit/a value]. Istio currently has no user)
[//]: # (story template. Maybe you can make one?)

[//]: # (User stories must be presented in a work group meeting. They need no approval and are later integrated)
[//]: # (into the RFCs, which do need approval for alpha. You may find value to negotiate within the work group where the)
[//]: # (user stories are presented to help clarify the user stories.)

- [x] [RFC Authored] - https://docs.google.com/document/d/1xQdZsEgJ3Ld2qebfT3EJkg2COTtCR1TqBVojmnvI78g/edit

[//]: # (An RFC is mandatory to graduate to experimental. The RFC does not have to be reviewed in a work group)
[//]: # (meeting to graduate to experimental.)

- [ ] [Documentation](TBD, will update istio.io later) for enabling and using the feature.

[//]: # (The documentation instructions may exist on the developer wiki or the team drive. They may include instructions)
[//]: # (for building running a `istioctl experimental command`, or using the preview profile,)
[//]: # (or any other relevant information.)

- [ ] [Feedback plan](will collect feedback on github, discuss.istio.io and slack). 

[//]: # (This may include user feedback meetings, discuss.istio.io conversations, GitHub issues, or mailing lists.)

- [x] Disabled by default. (Requires specifying an annotation "istio.io/dry-run: true" on authorization policy to enable the feature)

- [x] No impact on performance when the feature is disabled. (No changes to current code flow when disabled)

**Comment by brian-avery** [comment]
FYI, moved this to https://github.com/brian-avery/enhancements/blob/migrateFeaturesToFiles/features/dry_run_authorization_policy.md in #26. Instead of tracking feature promotion in individual files for experimental/alpha/beta/stable for each feature, this uses a living document for each feature. The scaffolding is in the repo for this already. Feel free to copy the file from that PR.

**Comment by yangminzhu** [comment]
@brian-avery could you also move https://github.com/istio/enhancements/issues/23 to your PR? I will close this issue.

**Comment by brian-avery** [comment]
Sure. Thanks. My plan is to move everything that's currently open to the PR. Just didn't finish yet.
